pkg/cmd/cli/cmd/importimage.go

```diff
@@ -320,8 +320,12 @@ func (o *ImportImageOptions) createImageImport() (*imageapi.ImageStream, *imagea
 			}
 
 			if ok {
+				// disallow re-importing anything other than DockerImage
+				if existing.From != nil && existing.From.Kind != "DockerImage" {
+					return nil, nil, fmt.Errorf("tag %q points to existing %s %q, it cannot be re-imported", tag, existing.From.Kind, existing.From.Name)
+				}
 				// disallow changing an existing tag
-				if existing.From == nil || existing.From.Kind != "DockerImage" {
+				if existing.From == nil {
 					return nil, nil, fmt.Errorf("tag %q already exists - you must use the 'tag' command if you want to change the source to %q", tag, from)
```

Comment:
Yeah, but we lack information why you can't import it. One assumes if the tag is there it should be importable, which is not true for `kind != DockerImage`.